app/src/main/java/com/babylon/wallet/android/data/repository/cache/database/ResourceEntity.kt

```diff
@@ -156,33 +150,44 @@ data class ResourceEntity(
             }
             return from(
                 address = ResourceAddress.init(address),
-                metadataCollection = metadata,
+                explicitMetadata = explicitMetadata,
+                implicitMetadata = metadata,
                 details = details,
                 type = type,
                 synced = synced
             )
         }
 
+        @Suppress("LongParameterList")
         private fun from(
             address: ResourceAddress,
-            metadataCollection: EntityMetadataCollection?,
+            explicitMetadata: EntityMetadataCollection?,
+            implicitMetadata: EntityMetadataCollection?,
             details: StateEntityDetailsResponseItemDetails?,
             type: ResourceEntityType,
             synced: Instant
         ): ResourceEntity {
-            val metadata = metadataCollection?.toMetadata().orEmpty()
+            val metadataColumn = if (implicitMetadata != null) {
```

Comment:
what is the difference between explicit/implicit metadata?

Reply:
Explicit metadata are the ones we request when we communicate with GW and GW ensures that we will receive them. Also known as [standard](https://docs.radixdlt.com/docs/metadata-for-wallet-display) metadata. For example a resource may have defined 150 metadata. If you do a simple entity details request, on the first response we will only receive the first page of them (100) and we need to paginate them in order to get the other 50.

Although we need to somehow make sure that some metadata will be received since are crucial to rendering the UI without the need of pagination, like name, symbol, description, icon_url....

So for these requests you will see that we put `ExplicitMetadataKey.forAssets` in order to retrieve in the `explicit_metadata` field these crucial ones we need for ui.

When the user navigates to asset details, we can also display the rest of the metadata (besides the explicit ones). Decided to call them implicit. If we know that there is a page to be paginated, this is the time we do it.